File: Queues_ll.py
class Node:
	def __init__(self, data):
		self.data = data
		self.next = None

class Queue:
	def __init__(self):
		self.front = self.rear = None

	def Enqueue(self, item):
		newNode = Node(item)

		print(str(item) + ' added to queue')
		
		if self.rear == None:
			self.front = self.rear = newNode
			return 

		self.rear.next = newNode
		self.rear = newNode

	def Dequeue(self):
		if self.front == None:
			print("Queue is empty")
			return

		temp = self.front
		self.front = self.front.next
		print(str(temp.data) + ' removed from queue')

		if self.front == None:
			self.rear = None

if __name__ == '__main__':
	q = Queue()
	q.Enqueue(1)	
	q.Enqueue(2)
	q.Enqueue(3)

	q.Dequeue()
	q.Dequeue()
	q.Dequeue()
	q.Dequeue()

File: Queues_ll.c
#include<stdio.h>
#include<stdlib.h>
struct Qnode
{
	int key;
	struct Qnode *next;
};
struct Queue
{
	struct Qnode *front,*rear;
};
struct Qnode *createNode(int key)
{
	struct Qnode *qn = (struct Qnode*)malloc(sizeof(struct Qnode));
	qn -> key = key;
	qn -> next = NULL;
	return qn;
};
struct Queue *createQueue()
{
	struct Queue *q = (struct Queue*)malloc(sizeof(struct Queue));
	q->front = q->rear = NULL;
	return q;
};
void Enqueue(struct Queue* q, int key){
	struct Qnode *qn = createNode(key);
	printf("Enqueued item is %d \n", qn->key);
	if (q->rear == NULL)
	{
		q->front = q->rear = qn;
		return;
	}
	q->rear->next = qn;
	q->rear = qn;
}
void Dequeue(struct Queue *q){
	if(q->front == NULL)
		return;
	struct Qnode *temp = q->front;
	q->front = q->front->next;

	printf("Dequeued item is %d \n", temp->key);

	if(q->front == NULL)
		q->rear = NULL;
}
int main()
{
	struct Queue *q = createQueue();
	Enqueue(q,50);
	Enqueue(q,80);
	Enqueue(q,110);
	Enqueue(q,140);

	Dequeue(q);
	Dequeue(q);
	Dequeue(q);
	Dequeue(q);

	return 0;
}